File: src/main/java/com/moshi/common/interceptor/TimingInterceptor.java
package com.moshi.common.interceptor;

import com.jfinal.aop.Interceptor;
import com.jfinal.aop.Invocation;
import com.jfinal.log.Log;

import java.util.Date;

public class TimingInterceptor implements Interceptor {
  @Override
  public void intercept(Invocation inv) {
    long start = new Date().getTime();
    inv.invoke();
    long end = new Date().getTime();
    Log log = Log.getLog("timing");
    log.info("Timing " + inv.getControllerKey() + ": " + (end - start) + "ms");
  }
}

File: src/main/java/com/moshi/common/interceptor/TimingInterceptor.kt
package com.moshi.common.interceptor

import com.jfinal.aop.Interceptor
import com.jfinal.aop.Invocation
import com.jfinal.log.Log

import java.util.Date

class TimingInterceptor : Interceptor {


  override fun intercept(inv: Invocation) {
    val start = System.currentTimeMillis()
    inv.invoke()
    val end = System.currentTimeMillis()
    log.info(
      "Timing " + inv.controller.request.requestURI + ": " + (end - start) + "ms"
    )
  }

  companion object {
    val log = Log.getLog("timing")
  }
}
